server/src/main/java/org/opensearch/extensions/rest/RestInitializeExtensionAction.java

```diff
@@ -62,36 +70,73 @@ public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client
         String openSearchVersion = null;
         String minimumCompatibleVersion = null;
         List<ExtensionDependency> dependencies = new ArrayList<>();
+        Set<String> additionalSettingsKeys = extensionsManager.getAdditionalSettings()
+            .stream()
+            .map(s -> s.getKey())
+            .collect(Collectors.toSet());
 
-        try (XContentParser parser = request.contentParser()) {
-            parser.nextToken();
-            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
-            while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
-                String currentFieldName = parser.currentName();
-                parser.nextToken();
-                if ("name".equals(currentFieldName)) {
-                    name = parser.text();
-                } else if ("uniqueId".equals(currentFieldName)) {
-                    uniqueId = parser.text();
-                } else if ("hostAddress".equals(currentFieldName)) {
-                    hostAddress = parser.text();
-                } else if ("port".equals(currentFieldName)) {
-                    port = parser.text();
-                } else if ("version".equals(currentFieldName)) {
-                    version = parser.text();
-                } else if ("opensearchVersion".equals(currentFieldName)) {
-                    openSearchVersion = parser.text();
-                } else if ("minimumCompatibleVersion".equals(currentFieldName)) {
-                    minimumCompatibleVersion = parser.text();
-                } else if ("dependencies".equals(currentFieldName)) {
-                    ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.currentToken(), parser);
-                    while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
-                        dependencies.add(ExtensionDependency.parse(parser));
-                    }
+        Tuple<? extends MediaType, Map<String, Object>> unreadExtensionTuple = XContentHelper.convertToMap(
+            request.content(),
+            false,
+            request.getXContentType().xContent().mediaType()
+        );
+        Map<String, Object> extensionMap = unreadExtensionTuple.v2();
+
+        ExtensionScopedSettings extAdditionalSettings = new ExtensionScopedSettings(extensionsManager.getAdditionalSettings());
+
+        try {
+            // checking to see whether any required fields are missing from extension.yml file or not
```

Comment:
Since we don't have `extensions.yml` anymore
```suggestion
            // checking to see whether any required fields are missing from the request or not
```

Reply:
Changed the message

Reply:
Didn't push the change?

Reply:
I changed a reference to `extensions.yml` in another class and missed this one. I change this one now.